packages/astro/src/core/dev/restart.ts

```diff
@@ -40,17 +39,19 @@ function shouldRestartContainer(
 	if (restartInFlight) return false;
 
 	let shouldRestart = false;
+	const normalizedChangedFile = vite.normalizePath(changedFile);
 
 	// If the config file changed, reload the config and restart the server.
 	if (inlineConfig.configFile) {
-		shouldRestart = vite.normalizePath(inlineConfig.configFile) === vite.normalizePath(changedFile);
+		shouldRestart = vite.normalizePath(inlineConfig.configFile) === normalizedChangedFile;
 	}
 	// Otherwise, watch for any astro.config.* file changes in project root
 	else {
-		const normalizedChangedFile = vite.normalizePath(changedFile);
 		shouldRestart = configRE.test(normalizedChangedFile);
-
-		if (preferencesRE.test(normalizedChangedFile)) {
+		const settingsPath = vite.normalizePath(
+			fileURLToPath(new URL('settings.json', settings.dotAstroDir)),
+		);
+		if (settingsPath.match(normalizedChangedFile)) {
```

Comment:
What is this trying to do here? It's calling `str1.match(str2)`

Reply:
I tried a few things and it was not that easy. Basically, I think it's more reliable to use `dotAstroDir` in case it becomes configurable, or changes. But then, I had to find a way to make both paths match but some had to normalize stuff etc. Anyways, I tried a bunch of combinations but that's the only one I found was working. If you have suggestions, happy to apply them!

Reply:
Both are not exactly equal, I think one of them has a prefix like `file:///` or something

Reply:
I don't think it's possible for either to have `file://` from reading the code, but anyways it would be nice to move away from `.match` and use `.includes` or `.endsWith` if needed.